File: doc/cpp/examples/IO/CxSerialPort/CxSerialPort_01.cpp

#include "xie_core.h"
#include "xie_high.h"
#include <stdio.h>

namespace User
{

// ============================================================
void CxSerialPort_01()
{
	printf("%s\n", __FUNCTION__);

	xie::IO::CxSerialPort port1;
	xie::IO::CxSerialPort port2;
	xie::CxStopwatch watch;

	try
	{
		int seqno = 0;

		// PortName
		xie::CxString port1_name = "";
		xie::CxString port2_name = "";
#if defined(_MSC_VER)
		port1_name = "COM1";
		port2_name = "COM11";
#else
		port1_name = "/dev/ttyS1";
		port2_name = "/dev/ttyS2";
#endif

		// Param
		xie::IO::TxSerialPort param(
			115200,
			xie::IO::ExParity::None,
			8,
			xie::IO::ExStopBits::One,
			xie::IO::ExHandshake::None
			);

		port1.PortName(port1_name);
		port2.PortName(port2_name);
		port1.Param(param);
		port2.Param(param);
		port1.Setup();
		port2.Setup();

		// ----------------------------------------------------------------------
		printf("[%d] port1.Write()\n", ++seqno);
		{
			const char*	buf = "hello";
			const int	buflen = strlen(buf);

			watch.Start();
			int count = port1.Write(buf, buflen, 1000);
			watch.Stop();
			printf("[%d] port1.Write returned %d. [%s], %9.3f msec\n", seqno, count, buf, watch.Lap);
		}

		// ----------------------------------------------------------------------
		printf("[%d] port2.Read()\n", ++seqno);
		{
			const int	buflen = 256;
			char		buf[buflen] = {0};

			watch.Start();
			int count = port2.Read(buf, buflen, 1000);
			watch.Stop();
			printf("[%d] port2.Read returned %d. [%s], %9.3f msec\n", seqno, count, buf, watch.Lap);
		}
	}
	catch(const xie::CxException& ex)
	{
		printf("%s(%d): code=%d function=%s\n", ex.File(), ex.Line(), ex.Code(), ex.Function());
	}

	port1.Dispose();
	port2.Dispose();
}

}

File: doc/cli/examples/IO/CxSerialPort/CxSerialPort_01.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace User
{
	partial class Examples
	{
		[XIE.CxPluginExecute]
		public void CxSerialPort_01()
		{
			var port1 = new XIE.IO.CxSerialPort();
			var port2 = new XIE.IO.CxSerialPort();
			var param = new XIE.IO.TxSerialPort(115200, XIE.IO.ExParity.None, 8, XIE.IO.ExStopBits.One, XIE.IO.ExHandshake.None);

			// PortName
			string port1_name = "";
			string port2_name = "";
			switch (System.Environment.OSVersion.Platform)
			{
				case PlatformID.Unix:
					port1_name = "/dev/ttyS1";
					port2_name = "/dev/ttyS2";
					break;
				default:
					port1_name = "COM1";
					port2_name = "COM11";
					break;
			}

			try
			{
				// Parameter
				port1.PortName = port1_name;
				port1.Param = param;

				port2.PortName = port2_name;
				port2.Param = param;

				// Setup (Open Port)
				port1.Setup();
				port2.Setup();

				// Write
				{
					string text = "Hello, World!!";
					byte[] data = Encoding.ASCII.GetBytes(text);
					int length = port1.Write(data, data.Length, 1000);
					Console.WriteLine("port1.Write = {0} bytes", length);
				}

				// Read
				{
					byte[] data = new byte[256];
					int length = port2.Read(data, data.Length, 0);
					string text = Encoding.ASCII.GetString(data, 0, length);
					Console.WriteLine("port2.Read = {0} bytes, data=({1})", length, text);
				}
			}
			finally
			{
				// Dispose (Close Port)
				port1.Dispose();
				port2.Dispose();
			}
		}
	}
}
